prowler/providers/aws/services/ec2/ec2_service.py

```diff
@@ -11,6 +11,17 @@
 
 ################## EC2
 class EC2(AWSService):
+    check_ids = set()
```

Comment:
What about putting this into the `AWSService` class? 

I'm concerned about the logic to set failed checks:
- I'd rename `check_ids` to `failed_check_ids`
- Don't we need to store the resource id and region for it?

Reply:
> What about putting this into the `AWSService` class?

Yes, it could be used then also by other services as well, not only the ec2 service.

> I'm concerned about the logic to set failed checks:
> 
> * Don't we need to store the resource id and region for it?

You are right, thanks for the reminder. I adapted the code accordingly.

> * I'd rename `check_ids` to `failed_check_ids`

It's now renamed to `failed_checks` because it contains more than a `check_id` since I added the resource id and region as well.

Reply:
This makes a ton of sense!